temporal-sdk/src/main/java/io/temporal/activity/ActivityOptions.java

```diff
@@ -202,6 +204,22 @@ public Builder setCancellationType(ActivityCancellationType cancellationType) {
       return this;
     }
 
+    /**
+     * If set to true, will not request eager execution regardless of worker settings. If false,
+     * eager execution may still be disabled at the worker level or eager execution may not be
+     * requested due to lack of available slots.
+     *
+     * <p>Eager activity execution means the server returns requested eager activities directly from
+     * the workflow task back to this worker which is faster than non-eager which may be dispatched
+     * to a separate worker.
+     *
+     * <p>Default is false.
+     */
+    public Builder setDisableEagerExecution(boolean disableEagerExecution) {
```

Comment:
@bergundy In go-sdk, the activity level option for this is named `disableEagerExecution`. In TypeScript, you named it `allowEagerDispatch`, but kept the negative form later in the chain (the core-sdk use the negative form). Should I switch to the affirmative form _everywhere_? TBH that would make more sense in and would be easier to understand.

Reply:
In go absent struct member becomes the default value for that type which means that there’s no way to distinguish between absent and false and we want the default to be true, we don’t have that problem in TS

Reply:
It definitely should be disableEagerExecution, not enable. It's not about defaults, it's about the API naming and what is assumed to be default by the APIs. 

In java it's easy to distinguish between not set or false, but let's leave the `disableEagerExecution` naming for the mentioned reason.

Reply:
I don't see why calling the option `allowEager{Execution|Dispatch}` and saying that the default is true is bad naming.
I like using the positive form, it avoids having double negation when used in conditions later.

I even find this hard to read: `setDisableEagerExecution` and it says the default is `false`, there's some ambiguity here. Does it mean the default is default is disabled or that disable is false?

At the end of the day it's @Spikhalskiy call so do whatever feels right to you.

Reply:
Ideally, every flag should have a false default value. It makes things easy and expected from the user side. It's a flag.
A flag that is true by default is a worse problem for an API than a double negation.

Reply:
Fair enough, I will keep it as `disableEagerExecution`.

However, I think it might be better to replace the "Default: false" by a more understandable, affirmative-form phrase... Something like "By default, eager execution is requested if possible."

Reply:
If you want, add it as a clarification, I'm all for it. But leave an explicit statement of the default value of the parameter too, please. It's the style across the whole SDK and these two styles target a bit different kinds of readers.